### PR TITLE
Prepare for release: Baseline Plugin

### PR DESCRIPTION
**Pull request by samiebee43**
image

Baseline plugin is completed, with backwards-compatible additions to the plugin system. The additions allow plugin programmers to write shorter simpler programs and exposes additional control over plots. For this plugin I use this to draw extents in two different places while using only one label per pair. Also the colors in this one are consistent, which makes it easier to use this plugin repeatedly.

The immediate mode is only from the perspective of the plugin writer, and the same objects are being instantiated under the hood. Drawing this way is fairly slow--true immediate mode drawing would make a significant performance boost.

This branch includes a light automated refactor which groups BusinessObjects into python submodules.

### CODE REVIEW

**Comment by mzieg** [comment]
When testing, I tried to set wavelength 842.61, and although the spinner shows two decimal places, it only seems to accept integral values.

File: enlighten/Plugins/PluginWorker.py [hunk 1]
```diff
@@ -98,19 +98,15 @@ def run(self):
             log.debug("PluginWorker[%s] sending request %d", module_name, request.request_id)
             response = None
             try:
-                response = plugin.process_request(request)
+                # make wavelenghts/wavenumbers available to *to_pixel functions
```
[the remark this answers is not in the document]
Comment (samiebee43) [reply enlighten/Plugins/PluginWorker.py:101]: wavelenghts -> wavelengths

**Comment by samiebee43** [comment]
I addressed most of what we talked about:

- fix non-clearing legend
- grant x-axis unit to plugin (and ctl ref)
- correct logic error in area_under_curve
- sample x at selected axis unit

I am not 100% sure about the exception that came up. It seems like it had to do with being in wavenumbers while using wavelengths_to_pixels.. but even then *to_pixel functions are supposed to _always_ return something.

The thing changed pretty significantly to where we are always using the selected unit to pixels before indexing-- so the exception may be gone now. It looks good when I default to 10,000 locally
image

**Comment by samiebee43** [comment]
One more thing, setting the step to 0.01 interferes with the ability to move the chart using the scrollwheel or arrow keys. The solution could be to set it 0.2 or so

**Comment by samiebee43** [comment]
@mzieg Please merge to main so I can release Enlighten 4.0.7 (partial refactor, plugin refactor, immediate mode plugin shim, LocalBaseline plugin)

File: enlighten/measurement/MeasurementFactory.py [hunk 2]
```diff
@@ -7,19 +7,19 @@
...
-from .ThumbnailWidget  import ThumbnailWidget
-from .DashFileParser   import DashFileParser
-from .Measurement      import Measurement
+from enlighten.parser.ColumnFileParser import ColumnFileParser
+from enlighten.parser.ExportFileParser import ExportFileParser
+from enlighten.parser.TextFileParser import TextFileParser
+from enlighten.parser.SPCFileParser import SPCFileParser
+from enlighten.ui.ThumbnailWidget import ThumbnailWidget
+from enlighten.parser.DashFileParser import DashFileParser
+from enlighten.measurement import Measurement
```
Comment (mzieg) [inline enlighten/measurement/MeasurementFactory.py:16]: Pretty sure this should be:
`from enlighten.measurement.Measurement import Measurement`

**Pull request merged**